File: day 17/code.c
/**
 * Definition for singly-linked list.
 * struct  {
 *     int val;
 *     struct ListNode *next;
 * };
 */
void deleteNode(struct ListNode* node) {
    
    struct ListNode *temp = node->next;
    while ( temp->next != NULL )
    {
        node->val = temp->val;
        temp = temp->next;
        node = node->next;
    }
    node->val = temp->val;
    node->next = NULL;
    
}

File: day 22/code.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {   
  public int RecOddEven(ListNode odd, ListNode even)
  {
      if(odd.next == null || even.next == null)
      {
          odd.next = null;
          return 0;
      }
      odd.next =  odd.next.next;
      even.next = even.next.next;
      return RecOddEven(odd.next, even.next);
  }
  public ListNode oddEvenList(ListNode head) {
      if(head == null)
          return head;
      ListNode even = head.next , temp = head;
      RecOddEven(head,even);
      while(temp.next!= null)
          temp = temp.next;
      temp.next = even;
      return head;
  }
}